### PR TITLE
DOC: Update doc and internal links after uxl transfer

### PR DESCRIPTION
**Pull request by david-cortes-intel**
## Description

Similar to https://github.com/uxlfoundation/scikit-learn-intelex/pull/2203

After this repository got transferred to uxlfoundation, the doc links for github do not work anymore as they do not redirect to the new URL.

This PR changes all the instances of string "intel.github.io" to "uxlfoundation.github.io" which should be the new URL for these docs.

It also modifies internal links to files within the repository to point to `uxlfoundation/scikit-learn-intelex` instead.

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.

**Performance**

Not applicable.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -58,40 +58,40 @@ There are several options available for installing oneDAL:
...
 ## Documentation
 
 oneDAL documentation:
 
-- [Release Notes](https://github.com/oneapi-src/oneDAL/releases)
-- [Get Started Guide](https://oneapi-src.github.io/oneDAL/quick-start.html)
-- [Developer Guide and Reference](http://oneapi-src.github.io/oneDAL/)
+- [Release Notes](https://github.com/uxlfoundation/oneDAL/releases)
+- [Get Started Guide](https://uxlfoundation.github.io/oneDAL/quick-start.html)
+- [Developer Guide and Reference](http://uxlfoundation.github.io/oneDAL/)
```
Comment (Vika-F) [inline README.md:89]:
```suggestion
- [Developer Guide and Reference](https://uxlfoundation.github.io/oneDAL/)
```

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -28,8 +28,8 @@ cpp/           @Alexsandruss @samir-nasibli @Alexandr-Solovev
...
-/.github/CODEOWNERS @oneapi-src/oneDAL-maintainer
-/SECURITY.md @oneapi-src/oneDAL-maintainer
-/MAINTAINERS.md @oneapi-src/oneDAL-maintainer
-/CONTRIBUTING.md @oneapi-src/oneDAL-maintainer
-/CODE_OF_CONDUCT.md @oneapi-src/oneDAL-maintainer
+/.github/CODEOWNERS @uxlfoundation/oneDAL-maintainer
+/SECURITY.md @uxlfoundation/oneDAL-maintainer
+/MAINTAINERS.md @uxlfoundation/oneDAL-maintainer
+/CONTRIBUTING.md @uxlfoundation/oneDAL-maintainer
+/CODE_OF_CONDUCT.md @uxlfoundation/oneDAL-maintainer
```
Comment (homksei) [inline .github/CODEOWNERS:35]:
```suggestion
/.github/CODEOWNERS @uxlfoundation/oneDAL-maintain
/SECURITY.md @uxlfoundation/oneDAL-maintain
/MAINTAINERS.md @uxlfoundation/oneDAL-maintain
/CONTRIBUTING.md @uxlfoundation/oneDAL-maintain
/CODE_OF_CONDUCT.md @uxlfoundation/oneDAL-maintain
```

File: docs/source/examples.rst [hunk 3]
```diff
@@ -26,4 +26,4 @@ oneAPI Examples
    DPC++ <examples/dpc.rst>
    C++ <examples/cpp.rst>
 
-.. note:: Explore the complete list of oneAPI code samples in the `oneAPI Samples Catalog <https://oneapi-src.github.io/oneAPI-samples/>`_. These samples were designed to help you develop, offload, and optimize multiarchitecture applications targeting CPUs, GPUs, and FPGAs. 
+.. note:: Explore the complete list of oneAPI code samples in the `oneAPI Samples Catalog <https://uxlfoundation.github.io/oneAPI-samples/>`_. These samples were designed to help you develop, offload, and optimize multiarchitecture applications targeting CPUs, GPUs, and FPGAs.
```
Comment (homksei) [inline docs/source/examples.rst:29]: same thing https://github.com/uxlfoundation/scikit-learn-intelex/pull/2203#discussion_r1867842380

Comment (david-cortes-intel) [reply docs/source/examples.rst]: Thanks, reverted the change there.

**Review by homksei — APPROVED** [review]

**Review by ethanglaser — APPROVED** [review]

**Pull request merged**